src/jarabe/journal/journalactivity.py

```diff
@@ -282,17 +400,70 @@ def _setup_secondary_view(self):
         self._secondary_view.pack_end(self._detail_view, True, True, 0)
         self._detail_view.show()
 
+    def _key_press_add_new_entry_cb(self, window, event):
+        keyname = Gdk.keyval_name(event.keyval)
+        if keyname == 'Return':
+            self._add_new_button_clicked_cb(None, None)
+
+        return False
+
+    def _add_new_button_clicked_cb(self, button, event):
+        if self.get_list_view().get_projects_view_active():
+            logging.debug('[GSoC]for project title %r' %self.get_entry().props.text)
+            initialize_journal_object(title= self.get_entry().props.text,
+                                      bundle_id=PROJECT_BUNDLE_ID,
+                                      activity_id=None,
+                                      project_metadata=None)
+        elif self.project_metadata is not None:
+            chooser = ActivityChooser()
+            text = _("Choose an activity to start '%s' with" %self._entry_project.props.text)
+            chooser.set_title(text)
+            chooser.connect('activity-selected',self.__activity_selected_cb)
+            chooser.show_all()
+
+    def __activity_selected_cb(self, widget, bundle_id, activity_id):
+        initialize_journal_object(title = _(self._entry_project.props.text),
+                                        bundle_id=bundle_id, 
+                                        activity_id=activity_id,
+                                        project_metadata=self.project_metadata)
+
     def _key_press_event_cb(self, widget, event):
-        if not self._main_toolbox.search_entry.has_focus():
-            self._main_toolbox.search_entry.grab_focus()
+        #if not self._main_toolbox.search_entry.has_focus():
+        #self._main_toolbox.search_entry.grab_focus()
 
         keyname = Gdk.keyval_name(event.keyval)
         if keyname == 'Escape':
             self._main_toolbox.clear_query()
             self.show_main_view()
 
+    def __choose_project_cb(self, tree_view, metadata_to_copy):
+        project_chooser = ObjectChooser()
```

Comment:
I have set filter in line below
`project_chooser._toolbar._proj_list_button_clicked_cb(None)`